### PR TITLE
Add support for style merging in mergeProps

### PR DESCRIPTION
**Pull request by ryo-manba**
Closes https://github.com/adobe/react-spectrum/issues/2507

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

## 📝 Test Instructions:

Add the following code to Storybook and verify that styles are merged correctly.

```tsx
export const Example = () => {
  let stylePropsA = {height: '50px', width: '50px'};
  let stylePropsB =  {height: '75px', color: 'orange'};
  // Expected: {style: {height: 75px, width: 50px, color: orange}}
  let result = mergeProps({style: stylePropsA}, {style: stylePropsB});
  return (
    <div {...result}>
      example
    </div>
  );
};
```

### CODE REVIEW

File: packages/@react-aria/interactions/test/usePress.test.js [hunk 1]
```diff
@@ -21,9 +21,9 @@ import {theme} from '@react-spectrum/theme-default';
 import {usePress} from '../';
 
 function Example(props) {
-  let {elementType: ElementType = 'div', style, draggable, ...otherProps} = props;
+  let {elementType: ElementType = 'div', draggable, ...otherProps} = props;
   let {pressProps} = usePress(otherProps);
-  return <ElementType {...pressProps} style={style} tabIndex="0" draggable={draggable}>{ElementType !== 'input' ? 'test' : undefined}</ElementType>;
+  return <ElementType {...pressProps} tabIndex="0" draggable={draggable}>{ElementType !== 'input' ? 'test' : undefined}</ElementType>;
```
Comment (ryo-manba) [inline packages/@react-aria/interactions/test/usePress.test.js:26]: see: https://github.com/adobe/react-spectrum/pull/2457#discussion_r751442528

File: packages/@react-aria/utils/test/mergeProps.test.js [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 import clsx from 'clsx';
 import {mergeIds} from '../src/useId';
-import {mergeProps} from '../';
+import {mergeProps} from '../src/mergeProps';
```
Comment (ryo-manba) [inline packages/@react-aria/utils/test/mergeProps.test.js:15]: I changed the import path because the changes were not reflected after saving.

**Review by snowystinger — COMMENTED** [review]
Thanks for the PR. Unfortunately, this has messed with some of our VRT https://www.chromatic.com/test?appId=5f0dd5ad2b5fc10022a2e320&id=65ee2607b7324b7ea6d3dc47

fortunately, I think the fix should be mostly contained to https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/story-utils/src/GeneratePowerset.tsx

**Comment by devongovett** [comment]
I think one could argue this is a breaking change as well. At this point, people might be relying on the fact that styles are _not_ merged.

**Comment by snowystinger** [comment]
> I think one could argue this is a breaking change as well. At this point, people might be relying on the fact that styles are _not_ merged.

This is a good point. If it suddenly started merging, people could suddenly have extra styles coming through.

Do we actually want to do this? we could add an argument that describes how to merge the props, either a simple boolean for merge styles, or more complex, a method to give control of the merge...

Unless we do it by default though, it doesn't really help with instances such as https://github.com/adobe/react-spectrum/pull/2457#discussion_r750792753 because the user has to realize that there two competing style objects.

We'd discussed a version of mergeProps which does memo'ing, maybe we handle it there? *waves hands towards the future*

**Comment by snowystinger** [comment]
Going to close this one until we have more information on the issue. Thanks for understanding and thanks for starting it!

**Pull request closed without merging**